protocol/storage/client.go

```diff
@@ -308,16 +307,20 @@ type ClientNodeImpl struct {
 	dserv     ipld.DAGService
 	host      host.Host
 	blockTime time.Duration
-	*ping.PingService
+	plumbing  clientNodeImplPlumbing
+}
+
+type clientNodeImplPlumbing interface {
```

Comment:
blocking: sorry i'm confused about why we need to do this. i expected to see ping removed from clientNode above, and added to clientPorcelainAPI above, and the call on line 203 to smc.node.Ping changed to smc.porcelainapi.ping. that eliminates node as an intermediary for this dependency. cc @anorth

Reply:
`clientNodeImplPlumbing`'s dependency on Ping was introduced in a rebase. Originally, I had removed Ping from the node entirely as it wasn't needed anywhere porcelain wasn't available. After the dependency was added, I did this as an attempt to avoid re-adding porcelain to the node.

Now, I've put Ping back in the node and removed porcelain from `clientNodeImplPlumbing`.